packages/manager/src/features/CloudPulse/Alerts/Utils/utils.ts

```diff
@@ -118,3 +128,81 @@ export const getChipLabels = (
     };
   }
 };
+/**
+ * Filters the given alert's dimension filter values and returns a DimensionFilter object.
+ * @param formValues The alert dimension filter values from the form.
+ * @returns The filtered DimensionFilter object.
+ */
+export const convertAlertDefinitionDimensionFilterValues = ({
+  dimension_label,
+  operator,
+  value,
+}: AlertDefinitionDimensionFilter): DimensionFilter => {
+  return {
+    dimension_label,
+    operator,
+    value,
+  };
+};
+
+/**
+ * Filters the alert's metric values from the form and returns a MetricCriteria object.
+ * @param formValue The alert's metric criteria values from the form.
+ * @returns The filtered MetricCriteria object.
+ */
+export const convertAlertDefinitionMetricValues = ({
+  aggregate_function,
+  dimension_filters,
+  metric,
+  operator,
+  threshold,
+}: AlertDefinitionMetricCriteria): MetricCriteria => {
+  return {
+    aggregate_function,
+    dimension_filters:
+      dimension_filters?.map((filter) => {
+        return convertAlertDefinitionDimensionFilterValues(filter);
+      }) ?? [],
+    metric,
+    operator,
+    threshold,
+  };
+};
+
+/**
+ * Filters and maps the alert data to match the form structure.
+ * @param alert The alert object to be mapped.
+ * @param serviceType The service type for the alert.
+ * @returns The formatted alert values suitable for the form.
+ */
+export const convertAlertDefinitionValues = (
+  {
+    alert_channels,
+    description,
+    entity_ids,
+    id,
+    label,
+    rule_criteria,
+    severity,
+    tags,
+    trigger_conditions,
+  }: Alert,
+  serviceType: AlertServiceType
+): EditAlertPayloadWithService => {
+  return {
+    alertId: id,
+    channel_ids: alert_channels.map((channel) => channel.id),
+    description: description || undefined,
+    entity_ids,
+    label,
+    rule_criteria: {
+      rules: rule_criteria.rules.map((rule) =>
+        convertAlertDefinitionMetricValues(rule)
+      ),
+    },
```

Comment:
This makes it way more obvious that we're omitting the label property without all the fluff.
```suggestion
    rule_criteria: {
    rules: rule_criteria.rules.map(rule => ({
      ...rule,
      dimension_filters: rule.dimension_filters?.map(({ label, ...filter }) => filter) ?? []
    }))
  },
```